src/Storage/ClaimController.ts

```diff
@@ -79,23 +87,44 @@ export class ClaimController {
   }: {
     retryDelay?: number
     maxAttempts?: number
-  } = {}) {
-    this.logger.child({ method: 'downloadNextHash' })
-    try {
-      this.logger.trace('Downloading next entry')
-      const result = await asyncPipe(
-        this.findEntryToDownload,
-        this.updateEntryAttempts,
-        this.downloadEntryClaim,
-        this.updateEntryPairs,
-        this.publishEntryDownload
-      )({ retryDelay, maxAttempts })
-      this.logger.info('Successfully downloaded entry')
-      return result
-    } catch (error) {
-      if (error instanceof NoMoreEntriesException) return this.logger.trace(error.message)
-      this.logger.error(error)
+  } = {}): Promise<void> {
+    const logger = this.logger.child({ method: 'downloadNextHash' })
+
+    const updateEntryFailureReason = (ipfsHash: string, failureType: FailureType, failureReason: FailureReason) =>
```

Comment:
This is being used with await below. Should this be an async function?

Reply:
It's not necessary for a function that returns a promise to be defined as async. 

I was letting the type of the function be inferred since it's a one liner that runs a query, but if you guys think it's confusing maybe declaring it as `: Promise<whatever mongodb returns>` could do?

Reply:
Yup your right, don't think I noticed the implicit return while looking at Jesse's screen :D

I think we could remove the `await` from the function calls though since we don't need the response?

Reply:
Not sure which one you’re talking about but in general we need to await even if we’re not using the result so errors get thrown. On my phone atm so hard to get context.